### PR TITLE
Fixes to `chainlink admin` command

### PR DESCRIPTION
**Pull request by george-dorin**
-Added new error type `errForbidden` to distinguish between not logged in and insufficient permissions 
-Removed muliterr as error message and the JSON response body is the same and we were displaying the same error twice 
-Set `newrole` flag as required for `chrole` command
-Improved error messages
-Fixed typos in comments
-Fixed delete command errors
-Added full information on delete command
-Added headers to the 403 response

https://smartcontract-it.atlassian.net/browse/BCF-1802

### CODE REVIEW

File: core/web/user_controller.go [hunk 1]
```diff
@@ -117,9 +118,24 @@ func (c *UserController) UpdateRole(ctx *gin.Context) {
...
+	_, err := clsession.GetUserRole(request.NewRole)
+	if err != nil {
+		jsonAPIError(ctx, http.StatusBadRequest, errors.New("new role does not exist, possible options are 'admin', 'edit', 'run', 'view'"))
+		return
+	}
+
 	user, err := c.App.SessionORM().UpdateRole(request.Email, request.NewRole)
 	if err != nil {
-		jsonAPIError(ctx, http.StatusInternalServerError, errors.New("error updating API user"))
+		jsonAPIError(ctx, http.StatusInternalServerError, multierr.Combine(errors.New("error updating API user"), err))
```
[the remark this answers is not in the document]
Comment (george-dorin) [reply core/web/user_controller.go]: Fixed in https://github.com/smartcontractkit/chainlink/pull/8311/commits/4564ab9300a9e4c686f18be4c570cad79d26556a

**Review by jmank88 — DISMISSED** [review]

**Review by pinebit — APPROVED** [review]

File: core/cmd/remote_client.go [hunk 2]
```diff
@@ -330,10 +331,11 @@ func (cli *Client) parseResponse(resp *http.Response) ([]byte, error) {
 	if errors.Is(err, errUnauthorized) {
 		return nil, cli.errorOut(multierr.Append(err, fmt.Errorf("your credentials may be missing, invalid or you may need to login first using the CLI via 'chainlink admin login'")))
 	}
+	if errors.Is(err, errForbidden) {
+		return nil, cli.errorOut(multierr.Append(err, fmt.Errorf("this acction requires admin privileges current user does not have enough privileges to perform this action, login with admin via 'chainlink admin login'")))
```
Comment (essamhassan) [inline core/cmd/remote_client.go:335]: typo here

Comment (george-dorin) [reply core/cmd/remote_client.go]: Fixed in [5684118](https://github.com/smartcontractkit/chainlink/pull/8311/commits/5684118f7c94361b6ecfde1913486b064d8b496e)

File: core/cmd/remote_client.go [hunk 3]
```diff
@@ -330,10 +331,11 @@ func (cli *Client) parseResponse(resp *http.Response) ([]byte, error) {
 	if errors.Is(err, errUnauthorized) {
 		return nil, cli.errorOut(multierr.Append(err, fmt.Errorf("your credentials may be missing, invalid or you may need to login first using the CLI via 'chainlink admin login'")))
 	}
+	if errors.Is(err, errForbidden) {
+		return nil, cli.errorOut(multierr.Append(err, fmt.Errorf("this action requires admin privileges current user does not have enough privileges to perform this action, login with admin via 'chainlink admin login'")))
```
Comment (samsondav) [inline core/cmd/remote_client.go:335]:
```suggestion
		return nil, cli.errorOut(multierr.Append(err, fmt.Errorf("this action requires admin privileges. The current user does not have enough privileges to perform this action, login with admin via 'chainlink admin login'")))
```

Comment (samsondav) [inline core/cmd/remote_client.go:335]: Can we return the email of the currently logged in user here, for convenience?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/cmd/remote_client.go (hunk 3 above)
Comment (george-dorin) [reply core/cmd/remote_client.go]: Changed in [87d1067](https://github.com/smartcontractkit/chainlink/pull/8311/commits/87d1067a70e5b422e3763efba0f37954c7733777)

Comment (george-dorin) [reply core/cmd/remote_client.go]: Added in [87d1067](https://github.com/smartcontractkit/chainlink/pull/8311/commits/87d1067a70e5b422e3763efba0f37954c7733777)

**Review by jmank88 — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

**Pull request merged**